### PR TITLE
[Changelog] Add rollup abilities by product area

### PR DESCRIPTION
**Pull request by kodster28**
## Context

Often, folks have multiple products and/or want to consume the changes at a "platform" level. This is especially common for Zero Trust + Developer Platform.

This PR makes that possible by doing the following:
- Create specific landing pages that only show changes to a product area
- Create specific RSS feeds for ^^^ as well

To-dos:
- [x] Move the "logic" from the `changelog_file_name` parameter to a data-file specified "Platform Area" value
- [x] Add ^^^ as a filter to the main changelog page (likely will require some unique logic to populate the RSS feed link)
- [x] Fix desktop formatting (so it expands to fill more space on product area pages + date header is too far down)
- [x] Test for bugs / edge cases
- [ ] Add documentation to our style guide on how to use it

## Other notes

We have to approach creating area RSS pages by creating actual pages b/c of the limitations around creating dynamic pages w/in Hugo. If we don't have the specific landing pages, we won't actually be able to provide filtered RSS feeds.

### CODE REVIEW

File: content/cloudflare-one/changelog/_index.md [hunk 1]
```diff
@@ -4,8 +4,16 @@ title: Changelog
 meta:
     description: Review recent changes to Cloudflare Access.
 weight: 15
+layout: changelog
+changelog_name: Cloudflare Zero Trust
+changelog_product_area_name: Zero Trust
```
Comment (pedrosousa) [inline content/cloudflare-one/changelog/_index.md:9]: Since this is not an array, we could hide/remove the **Product Area** dropdown whenever this field is set (users can only select one value, and we're already filtering by that value).

File: content/cloudflare-one/changelog/casb.md [hunk 2]
```diff
@@ -13,6 +13,6 @@ outputs:
 
 # CASB Changelog
 
-<!-- All changelog entries live in /data/changelogs/access.yaml. For more details, refer to https://developers.cloudflare.com/style-guide/documentation-content-strategy/content-types/changelog/#yaml-file -->
+<!-- All changelog entries live in /data/changelogs/casbß.yaml. For more details, refer to https://developers.cloudflare.com/style-guide/documentation-content-strategy/content-types/changelog/#yaml-file -->
```
Comment (pedrosousa) [inline content/cloudflare-one/changelog/casb.md]:
```suggestion
<!-- All changelog entries live in /data/changelogs/casb.yaml. For more details, refer to https://developers.cloudflare.com/style-guide/documentation-content-strategy/content-types/changelog/#yaml-file -->
```

File: layouts/partials/wrangler-changelog.html [hunk 3]
```diff
@@ -23,14 +23,15 @@
...
-"description" (replaceRE `###[^-]+` "" $item.body)) -}}
+{{- $item_dict := (dict "publish_date" ($item.published_at | time.Format "2006-01-02")
+"title" (replaceRE `wrangler@` "" $item.name) "link" $item.html_url
+"description" (replaceRE `###[^-]+` "" $item.body) "product" "Wrangler") -}}
 {{- $results = $results | append $item_dict -}}
 {{- end -}}
 {{- end -}}
 
-{{- $combinedResult := (dict "wrangler" (dict "link" "/workers/platform/changelog/wrangler/" "productName" "Wrangler" "external" true
+{{- $combinedResult := (dict "wrangler" (dict "link" "/workers/platform/changelog/wrangler/" "productName" "Wrangler" "productArea" "Developer platform"
```
Comment (pedrosousa) [inline layouts/partials/wrangler-changelog.html]: Maybe we can define `$productArea` and `$productAreaLink` vars at the top of this file so that these hard-coded values are not "lost" in the middle of the file.

File: layouts/shortcodes/full-changelog.html [hunk 4]
```diff
@@ -43,21 +61,32 @@
...
+  </div>
   <div class="searchInput">
     <label for="searchInput">Text search</label>
     <input type="text" id="searchInput" placeholder="Search..." />
   </div>
 </div>
 
 <div id="changelogContent">
-{{- range $changelogRendered -}}
+{{- range $index, $entry := $changelogRendered -}}
```
Comment (pedrosousa) [inline layouts/shortcodes/full-changelog.html]:
```suggestion
{{- range $entry := $changelogRendered -}}
```
Nit: No need to instantiate an `$index` variable if we're not using it (it is optional).
This applies to other `range` occurrences.

File: layouts/shortcodes/full-changelog.html [hunk 5]
```diff
@@ -110,34 +141,52 @@ <h3>{{- .product -}}</h3>
...
 
   // Function to update description text
-
   function updateDescriptionText() {
     let productName = "Changelog";
+    let productAreaName = "Changelog";
     let changelogLink = "/changelog/index.xml";
+    let areaChangelogLink = "/changelog/index.xml";
     const regexPattern = `developers\.cloudflare\.com(.*?)#`;
+    productAreaRSSDescriptionText.style.display = "none"
```
Comment (pedrosousa) [inline layouts/shortcodes/full-changelog.html]:
```suggestion
    productAreaRSSDescriptionText.style.display = "none";
```

File: layouts/shortcodes/full-changelog.html [hunk 6]
```diff
@@ -110,34 +141,52 @@ <h3>{{- .product -}}</h3>
...
-    descriptionText.innerHTML = `Subscribe to all ${productName} posts via <a class='rssLink' target="_blank" href=${changelogLink}>RSS</a>.`;
+    for (const areaDropdown of productAreaDropdown) {
+      if (areaDropdown.selected === true && areaDropdown.value !== "all") {
+        productAreaName = areaDropdown.label;
+        dataRows.forEach((row) => {
+          if (row.getAttribute("data-product-area") === productAreaName.toLowerCase()) {
+              areaChangelogLink = row.getAttribute("data-product-area-link");
+              }
+          });
+          productAreaRSSDescriptionText.style.display = "block"
```
Comment (pedrosousa) [inline layouts/shortcodes/full-changelog.html]:
```suggestion
          productAreaRSSDescriptionText.style.display = "block";
```

File: layouts/shortcodes/full-changelog.html [hunk 7]
```diff
@@ -187,6 +240,14 @@ <h3>{{- .product -}}</h3>
       }
     }
   }
+  const productAreaFilter = urlParams.get("product-area");
+  if (productAreaFilter) {
+    for (const productArea of productAreaDropdown) {
+      if (productArea.value === productAreaFilter.toLowerCase()) {
+        productAreaDropdown.value = productAreaFilter.toLowerCase();
```
Comment (pedrosousa) [inline layouts/shortcodes/full-changelog.html:248]:
```suggestion
        productAreaDropdown.value = productAreaFilter.toLowerCase();
        break;
```
No need to continue iterating the `for` loop.

**Review by pedrosousa — APPROVED** [review]
